### PR TITLE
GEN-1673 - refact: make product reviews data available via ProductReviewsDataProvider

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Create `ProductReviewsDataProvider` to make product reviews data available whenever's needed.

## Justify why they are needed

Easier to work with as it's not a part of `ProductPageContext` anymore. In order to make _Product Reviews_ data available one just need to call `fetchProductReviewsData` (`memberReviews/productReviews.ts` file) server side and pass the result to `ProductReviewsDataProvider`.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3814?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3809** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3809?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3810** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3810?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3813** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3813?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #3814** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3814?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3814?utm_source=stack-comment).

File: apps/store/src/blocks/ProductAverageRatingBlock.tsx [hunk 1]
```diff
@@ -51,7 +52,7 @@ export const ProductAverageRatingBlock = () => {
 
               <Text color="textSecondaryOnGray" size={{ _: 'xs', md: 'sm' }}>
                 {/* TODO: lokalise this */}
-                Based on {averageRating.reviewCount} reviews
+                Based on {averageRating.totalOfReviews} reviews
```
Comment (robinandeer) [inline apps/store/src/blocks/ProductAverageRatingBlock.tsx:55]: not sure i follow the rename here? I kind of like the old name better hehe

File: apps/store/src/features/memberReviews/productReviews.ts [hunk 2]
```diff
@@ -1,21 +1,89 @@
 import { kv } from '@vercel/kv'
-import { AverageRating, ReviewComments } from './productReviews.types'
+import { z } from 'zod'
+import { getProductReviewsDistribution } from './getProductReviewsDistribution'
+import {} from './productReviews.types'
```
Comment (robinandeer) [inline apps/store/src/features/memberReviews/productReviews.ts:4]: this looks weird....

**Review by robinandeer — APPROVED** [review]
not sure i follow with all the changes but overall i think it looks good

File: apps/store/src/features/memberReviews/productReviews.ts (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/features/memberReviews/productReviews.ts]: Was forgotten there. It's removed now.

**Pull request merged**